server/src/main/java/io/crate/execution/engine/sort/Sort.java

```diff
@@ -57,13 +57,16 @@ public static <T> CompletableFuture<List<T>> parallelSort(List<T> list,
         return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
             .thenApply(aVoid -> {
 
-                ArrayList<T> result = new ArrayList<>(list.size());
+                // Re-using list to store sorted result.
+                // Saves NUM_BYTES_OBJECT_REF * list.size() bytes.
+                // Important: Parameter list is not used outside and thus can be mutated.
+                list.clear();
```

Comment:
This is not much: couple of Megabytes per million and comes with this not nice requirement to not use list outside (only pass it). I think not worth it and can be reverted